design-documents/graph-ql/coverage/gift-wrapping.md

```diff
@@ -0,0 +1,194 @@
+# Queries
+
+Gift message is Open source functionality and should be implemented in scope of GiftMessageGraphQl module.
+Gift wrapping is commerce functionality and should be covered in scope of GiftWrappingGraphQl module. 
+
+## Data
+
+```graphql
+
+###### Begin: Extending existing types ######
+type Cart {
+    available_gift_wrappings: [GiftWrapping]!
+    gift_wrapping: GiftWrapping
+    include_printed_card: Boolean! @doc(description: "Wether customer requested printed card for the order")
+    include_gift_receipt: Boolean! @doc(description: "Wether customer requested gift receipt for the order")
+    gift_message: GiftMessage
+}
+
+type SimpleCartItem {
+    available_gift_wrapping: [GiftWrapping]!
+    gift_wrapping: GiftWrapping
+    gift_message: GiftMessage
+}
+
+type ConfigurableCartItem {
+    available_gift_wrapping: [GiftWrapping]!
+    gift_wrapping: GiftWrapping
+    gift_message: GiftMessage
+}
+
+type BundleCartItem {
+    available_gift_wrapping: [GiftWrapping]!
+    gift_wrapping: GiftWrapping
+    gift_message: GiftMessage
+}
+
+type GiftCardCartItem {
+    gift_message: GiftMessage
+}
+
+type SalesItemInterface {
+    gift_wrapping: GiftWrapping
+    gift_message: GiftMessage
+}
+
+type CustomerOrder {
+    gift_wrapping: GiftWrapping
+    include_printed_card: Boolean! @doc(description: "Wether customer requested printed card for the order")
+    include_gift_receipt: Boolean! @doc(description: "Wether customer requested gift receipt for the order")
+    gift_message: GiftMessage
+}
+
+type CartPrices {
+    gift_options: GiftOptionsPrices
+}
+###### End: Extending existing types ######
+
+
+###### Begin: Defining new types ######
+type GiftWrapping {
+    id: ID!
+    design: String!
+    price: Money!
+    image: GiftWrappingImage!
+}
+
+type GiftWrappingImage {
+    label: String!
+    url: String!
+}
+
+type GiftOptionsPrices {
+    gift_wrapping_for_order: Money
+    gift_wrapping_for_items: Money
+    printed_card: Money
+}
+
+type GiftMessage {
+    to: String!
```

Comment:
Could you add doc blocks? Will it be a recipient email or name?

Reply:
Done.